metadata/src/main/java/org/apache/kafka/image/loader/MetadataBatchLoader.java

```diff
@@ -143,7 +143,8 @@ public long loadBatch(Batch<ApiMessageAndVersion> batch, LeaderAndEpoch leaderAn
             // 1) this is not the first record in this batch
             // 2) this is not the first batch since last emitting a delta
             if (transactionState == TransactionState.STARTED_TRANSACTION && (indexWithinBatch > 0 || numBatches > 0)) {
-                MetadataProvenance provenance = new MetadataProvenance(lastOffset, lastEpoch, lastContainedLogTimeMs);
+                // Accumulated delta is aligned on batch boundaries iff the BeginTransactionRecord is the first record in a batch
+                MetadataProvenance provenance = new MetadataProvenance(lastOffset, lastEpoch, lastContainedLogTimeMs, indexWithinBatch == 0);
```

Comment:
We should update the debug log a few lines below this.

From 
> handleCommit: Generated a metadata delta between 100 and 120 from 2 batch(es) in 4 us

To
> handleCommit: Generated an unaligned metadata delta between 100 and 120 from 2 batch(es) in 4 us

or similar